src/code/z_parameter.c

```diff
@@ -1828,7 +1828,7 @@ u8 Item_Give(PlayState* play, u8 item) {
         }
     } else if ((item >= ITEM_WEIRD_EGG) && (item <= ITEM_CLAIM_CHECK)) {
         if (item == ITEM_POACHERS_SAW) {
-            SET_ITEMGETINF(ITEMGETINF_1F);
+            SET_ITEMGETINF(ITEMGETINF_FOREST_STAGE_NUT_UPGRADE);
```

Comment:
Think its gonna need much more than just a comment in the code on where its used.
The flag name itself for example, should it include the name for both unrelated usages? probably. and theres gonna need to be a comment explaining the overlap on the flag definition itself.

Reply:
It is worth noting that this flag is never read for trade quest related reasons so I think naming it `ITEMGETINF_FOREST_STAGE_NUT_UPGRADE` and having a `@bug` comment saying that the flag shouldn't be set there is enough.
Although it does make me wonder what kind of developer experience setup they had for this mistake to be made.

Reply:
Something more along these lines
```c
// This flag is shared by two events. It is set when obtaining the Deku Nut Upgrade at the Forest Stage and when obtaining Poachers Saw. 
// This will make obtaining the Deku Nut upgrade impossible if Poachers Saw is obtained first.
// This flag is never read for the Poachers Saw event, so the overlap only causes an issue for the Deku Nut Upgrade. It will not prevent issues obtaining Poachers Saw.
#define ITEMGETINF_FOREST_STAGE_NUT_UPGRADE 0x1F
```